src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -57,11 +59,82 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
+                    controllerType,
+                    propertyName: property.Name);
+
+                if (metadata != null)
+                {
+                    UpdateParameterMetadata(metadata, property.BinderMetadata);
+                    propertyMetadata.Add(metadata);
+                }
+            }
+
+            var operationBindingContext = GetOperationBindingContext(actionContext, actionBindingContext);
+           
+            var controllerProperties = new Dictionary<string, object>(StringComparer.Ordinal);
+            await PopulateArgumentAsync(
+                operationBindingContext,
+                actionContext.ModelState,
+                controllerProperties,
+                propertyMetadata);
+            ActivateProperties(controller, controllerType, controllerProperties);
+
             var actionArguments = new Dictionary<string, object>(StringComparer.Ordinal);
-            await PopulateArgumentAsync(actionContext, actionBindingContext, actionArguments, parameterMetadata);
+            await PopulateArgumentAsync(
+                operationBindingContext,
+                actionContext.ModelState,
+                actionArguments,
+                parameterMetadata);
             return actionArguments;
         }
 
+        // Internal for tests
+        internal static ModelBindingContext GetModelBindingContext(
+            ModelMetadata modelMetadata,
+            ModelStateDictionary modelState,
+            OperationBindingContext operationBindingContext)
+        {
+            var modelBindingContext = new ModelBindingContext
+            {
+                ModelName = modelMetadata.BinderModelName ?? modelMetadata.PropertyName,
+                ModelMetadata = modelMetadata,
+                ModelState = modelState,
+
+                // Fallback only if there is no explicit model name set.
+                FallbackToEmptyPrefix = modelMetadata.BinderModelName == null,
+                ValueProvider = operationBindingContext.ValueProvider,
+                OperationBindingContext = operationBindingContext,
+            };
+
+            return modelBindingContext;
+        }
+
+        private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
+        {
+            foreach (var property in properties)
+            {
+                SetProperty(property.Key, containerType, controller, property.Value);
+            }
+        }
+
+        private void SetProperty(string propertyName, Type containerType, object container, object value)
+        {
+            var bindingFlags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.IgnoreCase;
+            var property = containerType.GetProperty(propertyName, bindingFlags);
+            if (property == null || !property.CanWrite)
+            {
+                // nothing to do
+                return;
+            }
```

Comment:
Should we filter these out earlier? Should we throw earlier? (we prob want to keep this check here as well). Should we have a model state error?